source/dub/packagesuppliers/fallback.d

```diff
@@ -40,19 +39,33 @@ private template fallback(T, alias func)
 {
 	import std.format : format;
 	enum fallback = q{
-		import std.range : back, dropBackOne;
-		import dub.internal.vibecompat.core.log : logDebug;
-		scope (failure)
+		import dub.internal.vibecompat.core.log : logDiagnostic;
+
+		Exception firstEx;
+		try
+			return m_suppliers[0].%1$s(args);
+		catch (Exception e)
+		{
+			logDiagnostic("Package supplier %%s failed with '%%s', trying fallbacks.",
+				m_suppliers[0].description, e.msg);
+			m_suppliers = m_suppliers[1 .. $];
+			firstEx = e;
```

Comment:
`-dip1008` won't allow this, but lots of stuff in Phobos does similar stuff which is why it will take a very long time until that switch gets ever activated ...

Reply:
I think ignoring that use-case (i.e. [`collectExceptions`](https://dlang.org/library/std/exception/collect_exception.html)) is reason enough for DIP1008 to never work until it's accompanied by a `.dup`/`.clone` mechanism.